databricks/koalas/frame.py

```diff
@@ -10049,6 +10049,64 @@ def explode(self, column):
         internal = self._internal.copy(spark_frame=sdf, data_spark_columns=data_scols)
         return DataFrame(internal)
 
+    def mad(self, axis=0):
+        """
+        Return the mean absolute deviation of values.
+
+        Parameters
+        ----------
+        axis : {index (0), columns (1)}
+            Axis for the function to be applied on.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'a': [1, 2, 3, np.nan], 'b': [0.1, 0.2, 0.3, np.nan]},
+        ...                   columns=['a', 'b'])
+
+        >>> df.mad()
+        a    0.666667
+        b    0.066667
+        Name: 0, dtype: float64
+
+        >>> df.mad(axis=1)
+        0    0.45
+        1    0.90
+        2    1.35
+        3     NaN
+        Name: 0, dtype: float64
+        """
+        from databricks.koalas import Series
+
+        axis = validate_axis(axis)
+
+        # Here we execute with the first 1000 to get the return type.
+        # If the records were less than 1000, it uses pandas API directly for a shortcut.
+        limit = get_option("compute.shortcut_limit")
+        pdf = self.head(limit + 1)._to_internal_pandas()
+        pser = pdf.mad(axis=axis)
+        if len(pdf) <= limit:
+            return Series(pser)
+
+        if axis == 0:
```

Comment:
I think we don't need to infer the return type when `axis=0`.

Maybe we should run this lines when only `axis=1`.

Reply:
Anyway, I think the line 10082 to 10108 is duplicated with `_reduce_for_stat_function`, we should duplicate it or implement this method without using `_reduce_for_stat_function` like you did in `Series.mad`

Reply:
Thanks, I applied it.